pkg/kubelet/kubelet_pods.go

```diff
@@ -1245,6 +1255,36 @@ func (kl *Kubelet) HandlePodCleanups(ctx context.Context) error {
 	return nil
 }
 
+// Filters terminal pods with deletion timestamp which are not runtime pods
+func (kl *Kubelet) filterTerminalPodsToDelete(allPods []*v1.Pod, runningRuntimePods []*kubecontainer.Pod, workingPods map[types.UID]PodWorkerSync) map[types.UID]*v1.Pod {
+	terminalPodsToDelete := make(map[types.UID]*v1.Pod)
+	for _, pod := range allPods {
+		if pod.DeletionTimestamp == nil {
+			// skip pods which don't have a deletion timestamp
+			continue
+		}
+		if !kl.isAdmittedPodTerminal(pod) {
+			// skip non-terminal pods
+			continue
+		}
+		if _, knownPod := workingPods[pod.UID]; knownPod {
+			// skip pods known to pod workers
+			continue
+		}
+		if _, knownPod := kl.statusManager.GetPodStatus(pod.UID); !knownPod {
```

Comment:
Calling GetPodStatus twice (once in isAdmittedPodTerminal and once here) feels wierd.  The logic in isAdmittedPodTerminal doesn't feel special enough to be called.

Reply:
In general reconciliation loops should make one call to other components to resynchronize their state - when we do lots of incremental calls it tends to confuse the process.

Reply:
Yeah, no need to call GetPodStatus twice. I think I will inline the `isAdmittedPodTerminal`, then remove this explicit check here.